src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -308,6 +321,66 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                         }
                     }
 
+                    break;
+                case TableExpressionKind.Include:
+                    var includeExpression = (IncludeExpression)tableExpression.NormalizedPredicate;
+
+                    StringBuilder.Append("SELECT Distinct ");
+                    StringBuilder.Append(V1.Resource.ResourceSurrogateId, "r").AppendLine(" AS Sid1, 0 as isMatch")
+                        .Append("FROM ").Append(V1.ReferenceSearchParam).Append(' ').AppendLine(referenceTableAlias)
+                        .Append("INNER JOIN ").Append(V1.Resource).Append(' ').AppendLine(resourceTableAlias);
+
+                    using (var delimited = StringBuilder.BeginDelimitedOnClause())
+                    {
+                        delimited.BeginDelimitedElement().Append(V1.ReferenceSearchParam.ReferenceResourceTypeId, referenceTableAlias)
+                            .Append(" = ").Append(V1.Resource.ResourceTypeId, resourceTableAlias);
+
+                        delimited.BeginDelimitedElement().Append(V1.ReferenceSearchParam.ReferenceResourceId, referenceTableAlias)
+                            .Append(" = ").Append(V1.Resource.ResourceId, resourceTableAlias);
+                    }
+
+                    using (var delimited = StringBuilder.BeginDelimitedWhereClause())
+                    {
+                        if (!includeExpression.WildCard)
+                        {
+                            delimited.BeginDelimitedElement().Append(V1.ReferenceSearchParam.SearchParamId, referenceTableAlias)
+                                .Append(" = ").Append(Parameters.AddParameter(V1.ReferenceSearchParam.SearchParamId, Model.GetSearchParamId(includeExpression.ReferenceSearchParameter.Url)));
+
+                            delimited.BeginDelimitedElement().Append(V1.ReferenceSearchParam.ReferenceResourceTypeId, referenceTableAlias)
+                                .Append(" = ").Append(Parameters.AddParameter(V1.ReferenceSearchParam.ReferenceResourceTypeId, Model.GetResourceTypeId(includeExpression.TargetResourceType)));
+                        }
+
+                        AppendHistoryClause(delimited, resourceTableAlias);
+                        AppendHistoryClause(delimited, referenceTableAlias);
+
+                        delimited.BeginDelimitedElement().Append(V1.ReferenceSearchParam.ResourceTypeId, referenceTableAlias)
+                            .Append(" = ").Append(Parameters.AddParameter(V1.ReferenceSearchParam.ResourceTypeId, Model.GetResourceTypeId(includeExpression.ResourceType)));
+
+                        // Limit the join to the main select CTE
+                        delimited.BeginDelimitedElement().Append(V1.Resource.ResourceSurrogateId, referenceTableAlias)
+                            .Append(" IN (SELECT TOP(")
+                            .Append(Parameters.AddParameter(context.MaxItemCount))
```

Comment:
why the top clause here?

Reply:
We're doing an `IN` from the top statement which has 11 items in it. We only want to include resources that are in the top 10 items since the 11th isn't included in the result set.

Reply:
Got it. Might be worth adding a comment.